### PR TITLE
Add a function wrapper for Jacobi polynomials.

### PR DESCRIPTION
**Pull request by skandalaCLIMA**
Add a function for computing Jacobi polynomials.
### Description

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

File: src/Numerics/Mesh/Elements.jl [hunk 1]
```diff
@@ -115,4 +115,45 @@ function interpolationmatrix(
...
+                a[3] =
+                    (2 * i + α + β - 2) * (2 * i + α + β - 1) * (2 * i + α + β)
+                a[4] = 2 * (i + α - 1) * (i + β - 1) * (2 * i + α + β)
+
+                V[:, i + 1] .=
+                    ((a[2] .+ a[3] .* x) .* V[:, i] .- a[4] .* V[:, i - 1]) ./
+                    a[1]
+            end
+        end
+    end
```
Comment (charleskawczynski) [inline src/Numerics/Mesh/Elements.jl:155]:
```suggestion
    V .= 0
    V[:, 1] .= 1

    np > 0 || return V
    V[:, 2] .= FT(0.5) .* (α .- β .+ (α .+ β .+ 2) .* x)
    np > 1 || return V
    for i in 2:np
        a[1] = (2 * i) * (i + α + β) * (2 * i + α + β - 2)
        a[2] = (2 * i + α + β - 1) * (α * α - β * β)
        a[3] = (2 * i + α + β - 2) * (2 * i + α + β - 1) * (2 * i + α + β)
        a[4] = 2 * (i + α - 1) * (i + β - 1) * (2 * i + α + β)

        num = ((a[2] .+ a[3] .* x) .* V[:, i] .- a[4] .* V[:, i - 1])
        V[:, i + 1] .= num ./ a[1]
    end
```

**Comment by charleskawczynski** [comment]
I was literally just looking into that

**Comment by skandalaCLIMA** [comment]
Ok. I am closing the PR.

**Comment by jkozdon** [comment]
>     a, b = GaussQuadrature.legendre_coefs(T, N)

Actually you want: `a, b = GaussQuadrature.jacobi_coefs(N, α, β)`

**Comment by skandalaCLIMA** [comment]
Do we currently have an test for Legendre polynomials? 
The polynomials computed using the above procedure, seems to scale differently compared to the standard Legendre polynomials.

**Comment by jkozdon** [comment]
> Do we currently have an test for Legendre polynomials?

Why do we need a test for something coming from another package?

What are you trying to do with them? I believe that the ones coming out of `GaussQuadrature.jl` are the orthonormal versions
```julia
using GaussQuadrature
using LinearAlgebra

N = 3
T = Float64

α = T(0)
β = T(0)
a, b = GaussQuadrature.jacobi_coefs(3, α, β)

# Use Legengre-Gauss to show orthogonality
r_lg, w_lg = GaussQuadrature.legendre(T, N + 1, GaussQuadrature.neither)
V_lg = GaussQuadrature.orthonormal_poly(r_lg, a, b)

# Check that this are orthonormal
@assert V_lg' * Diagonal(w_lg) * V_lg ≈ I

# Covert to form which starts with `1` are the end points
S = GaussQuadrature.orthonormal_poly([T(1)], a, b)
W_lg = V_lg ./ S

# Use show equivalence with Legengre-Gauss-Lobatto
r_lgl, _ = GaussQuadrature.legendre(T, N + 1, GaussQuadrature.both)
V_lgl = GaussQuadrature.orthonormal_poly(r_lgl, a, b)
W_lgl = V_lgl ./ S
```

output:

```julia
julia> include("try.jl")
4×4 Array{Float64,2}:
 1.0  -1.0        1.0  -1.0
 1.0  -0.447214  -0.2   0.447214
 1.0   0.447214  -0.2  -0.447214
 1.0   1.0        1.0   1.0
```

**Comment by skandalaCLIMA** [comment]
> > Do we currently have an test for Legendre polynomials?
> 
> Why do we need a test for something coming from another package?
> 
> What are you trying to do with them? I believe that the ones coming out of `GaussQuadrature.jl` are the orthonormal versions
> 
> ```julia
> using GaussQuadrature
> using LinearAlgebra
> 
> N = 3
> T = Float64
> 
> α = T(0)
> β = T(0)
> a, b = GaussQuadrature.jacobi_coefs(3, α, β)
> 
> # Use Legengre-Gauss to show orthogonality
> r_lg, w_lg = GaussQuadrature.legendre(T, N + 1, GaussQuadrature.neither)
> V_lg = GaussQuadrature.orthonormal_poly(r_lg, a, b)
> 
> # Check that this are orthonormal
> @assert V_lg' * Diagonal(w_lg) * V_lg ≈ I
> 
> # Covert to form which starts with `1` are the end points
> S = GaussQuadrature.orthonormal_poly([T(1)], a, b)
> W_lg = V_lg ./ S
> 
> # Use show equivalence with Legengre-Gauss-Lobatto
> r_lgl, _ = GaussQuadrature.legendre(T, N + 1, GaussQuadrature.both)
> V_lgl = GaussQuadrature.orthonormal_poly(r_lgl, a, b)
> W_lgl = V_lgl ./ S
> ```
> 
> output:
> 
> ```julia
> julia> include("try.jl")
> 4×4 Array{Float64,2}:
>  1.0  -1.0        1.0  -1.0
>  1.0  -0.447214  -0.2   0.447214
>  1.0   0.447214  -0.2  -0.447214
>  1.0   1.0        1.0   1.0
> ```

Thanks, Jeremy. I just added the corresponding wrapper function and test.

**Comment by skandalaCLIMA** [comment]
bors try

File: src/Numerics/Mesh/Elements.jl [hunk 2]
```diff
@@ -115,4 +115,21 @@ function interpolationmatrix(
...
+function jacobip(
+    α::FT,
+    β::FT,
+    np::Int,
+    x::AbstractArray{FT, 1},
+) where {FT <: AbstractFloat}
+    a, b = GaussQuadrature.jacobi_coefs(np, α, β)
+    V = GaussQuadrature.orthonormal_poly(x, a, b)
+    return V
+end
```
Comment (jkozdon) [inline src/Numerics/Mesh/Elements.jl:133]:
```suggestion
"""
    jacobip(α, β, N, x)

Returns a `(nx, np+1)` array containing the `N+1` Jacobi polynomials of order `N`, 
with parameter `(α, β)`, evaluated on 1D grid `x`.
"""
function jacobip( α, β, N, x)
    a, b = GaussQuadrature.jacobi_coefs(N, α, β)
    V = GaussQuadrature.orthonormal_poly(x, a, b)
    return V
end
```
elsewhere we use `N` for polynomial order. Type information is not needed

File: test/Numerics/Mesh/Elements.jl [hunk 3]
```diff
@@ -101,3 +102,23 @@ end
         @test D * P6(r) ≈ DP6(r)
     end
 end
+
+@testset "Jacobip" begin
+    for T in (Float32, Float64, BigFloat)
+        let
+            α, β, np = T(0), T(0), Int(3) # α, β (for Legendre polynomials) & polynomial order
+            x, wt = Elements.lglpoints(T, np + 1) # lgl points for polynomial order np
+            V = Elements.jacobip(α, β, np, x)
```
Comment (jkozdon) [inline test/Numerics/Mesh/Elements.jl:111]:
```suggestion
            α, β, N = T(0), T(0), 3 # α, β (for Legendre polynomials) & polynomial order
            x, wt = Elements.lglpoints(T, N + 1) # lgl points for polynomial order np
            V = Elements.jacobip(α, β, N, x)
```

File: test/Numerics/Mesh/Elements.jl [hunk 4]
```diff
@@ -101,3 +102,23 @@ end
...
+            # compare with orthonormalized exact solution
+            # https://en.wikipedia.org/wiki/Legendre_polynomials
+            V_exact = similar(V)
+            V_exact[:, 1] .= T(1)
+            V_exact[:, 2] .= x
+            V_exact[:, 3] .= T(0.5) * (3 * x .^ 2 .- T(1))
+            V_exact[:, 4] .= T(0.5) * (5 * x .^ 3 .- 3 * x)
```
Comment (jkozdon) [inline test/Numerics/Mesh/Elements.jl:118]:
```suggestion
            V_exact[:, 1] .= 1
            V_exact[:, 2] .= x
            V_exact[:, 3] .= (3 * x .^ 2 .- 1) / 2
            V_exact[:, 4] .= (5 * x .^ 3 .- 3 * x) / 2
```

File: test/Numerics/Mesh/Elements.jl [hunk 5]
```diff
@@ -101,3 +102,23 @@ end
...
+            scale = T(1) ./ sqrt.(diag(V_exact' * Diagonal(wt) * V_exact))
```
Comment (jkozdon) [inline test/Numerics/Mesh/Elements.jl:119]:
```suggestion
            scale = 1 ./ sqrt.(diag(V_exact' * Diagonal(wt) * V_exact))
```

File: src/Numerics/Mesh/Elements.jl (hunk 2 above)
Comment (skandalaCLIMA) [reply src/Numerics/Mesh/Elements.jl]: revised.

File: test/Numerics/Mesh/Elements.jl (hunk 5 above)
Comment (skandalaCLIMA) [reply test/Numerics/Mesh/Elements.jl]: revised.

**Review by simonbyrne — APPROVED** [review]

**Pull request merged**